src/Workspaces/Remote/ServiceHub/Host/RemoteAnalyzerAssemblyLoaderService.cs

```diff
@@ -16,16 +18,43 @@ namespace Microsoft.CodeAnalysis.Remote.Diagnostics
     [ExportWorkspaceService(typeof(IAnalyzerAssemblyLoaderProvider), WorkspaceKind.RemoteWorkspace), Shared]
     internal sealed class RemoteAnalyzerAssemblyLoaderService : IAnalyzerAssemblyLoaderProvider
     {
-        private readonly DefaultAnalyzerAssemblyLoader _loader = new DefaultAnalyzerAssemblyLoader();
-        private readonly ShadowCopyAnalyzerAssemblyLoader _shadowCopyLoader = new ShadowCopyAnalyzerAssemblyLoader(Path.Combine(Path.GetTempPath(), "VS", "AnalyzerAssemblyLoader"));
+        private readonly RemoteAnalyzerAssemblyLoader _loader;
+        private readonly ShadowCopyAnalyzerAssemblyLoader _shadowCopyLoader;
 
         [ImportingConstructor]
         [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
         public RemoteAnalyzerAssemblyLoaderService()
         {
+            var baseDirectory = Path.GetDirectoryName(Path.GetFullPath(typeof(RemoteAnalyzerAssemblyLoader).GetTypeInfo().Assembly.Location));
+            Debug.Assert(baseDirectory != null);
+
+            _loader = new(baseDirectory);
+            _shadowCopyLoader = new(Path.Combine(Path.GetTempPath(), "VS", "AnalyzerAssemblyLoader"));
         }
 
         public IAnalyzerAssemblyLoader GetLoader(in AnalyzerAssemblyLoaderOptions options)
             => options.ShadowCopy ? _shadowCopyLoader : _loader;
+
+        // For analyzers shipped in Roslyn, different set of assemblies might be used when running
+        // in-proc and OOP e.g. in-proc (VS) running on desktop clr and OOP running on ServiceHub .Net6
+        // host. We need to make sure to use the ones from the same location as the remote.
+        private class RemoteAnalyzerAssemblyLoader : DefaultAnalyzerAssemblyLoader
```

Comment:
Don't this need tests added for it to verify that this fully works?

Reply:
You are absolutely right. However, the best way to test this (and anything related to .net6 servicehub host) is via integration tests, which I will enable once all the required bits are available on our test machines.